x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/IndicesAndAliasesResolver.java

```diff
@@ -194,12 +192,11 @@ ResolvedIndices resolveIndicesAndAliases(IndicesRequest indicesRequest, MetaData
         if (indicesRequest instanceof AliasesRequest) {
             //special treatment for AliasesRequest since we need to replace wildcards among the specified aliases too.
             //AliasesRequest extends IndicesRequest.Replaceable, hence its indices have already been properly replaced.
+            assert indicesRequest instanceof IndicesRequest.Replaceable;
```

Comment:
is this new assert needed? after all the following cast will fail if the request is not a Replaceable...

Reply:
@javanna  I am being extra careful for when `AliasesRequest` might not extend `IndicesRequest.Replaceable` . They are both interfaces (so the work is minimal) so it might happen, I guess (I haven't tested).
Are you ok with keeping the assertion?

Reply:
but doesn't AliasesRequest extend IndicesRequest.Replaceable ? I think I don't follow. Not a big deal though :)